File: Algo/1137_Nth_Tribonacci_Number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int64_t a = 0, b = 1, c = 1, tmp;
        if (n == 0) {
            return 0;
        } else if (n <= 2) {
            return 1;
        }
        for (int i = 3; i <= n; ++i) {
            tmp = a + b + c;
            a = b;
            b = c;
            c = tmp;
        }
        return tmp;
    }
};

File: Algo/1137. N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        first = 0
        second = 1
        third = 1
        fourth = 2
        if n == 0:
            return 0
        elif n < 3:
            return 1
        for i in range(3, n + 1):
            fourth = first + second + third
            first = second
            second = third
            third = fourth
        return fourth
